Python error unit tests on delegation of backend_with_compiler_demo

Amyh11325 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#60689**

Summary:
Added a test for errors that occur with a compiler, specifically when an
operator is not supported by the backend.

Test Plan:
Running python test/test_jit.py TestBackendsWithCompiler -v returns a
success.

Reviewers:
myuan

Subscribers:

Tasks:
T91991412

Tags:

Differential Revision: [D29374513](https://our.internmc.facebook.com/intern/diff/D29374513)

Commit:
Python error unit tests on delegation of backend_with_compiler_demo

Summary:
Added a test for errors that occur with a compiler, specifically when an
operator is not supported by the backend.

Test Plan:
Running python test/test_jit.py TestBackendsWithCompiler -v returns a
success.

Reviewers:
myuan

Subscribers:

Tasks:
T91991412

Tags:

[ghstack-poisoned]

Amyh11325 commented:
@Amyh11325 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29374513).

iseeyuan commented on test/jit/test_backends.py:
```diff
@@ -562,6 +562,33 @@ def test_execution(self):
+        with self.assertRaisesRegexWithHighlight(RuntimeError,
+                                                 r"node of aten::mul is not supported in this compiler", ""):
+            scripted_module_n = torch.jit.script(ErrorMessagesWithCompiler.ModuleNotSupported())
+            lowered_module_n = torch._C._jit_to_backend("backend_with_compiler_demo", scripted_module_n, {"forward": {"": ""}})
```
Could we add more expected message here, especially,
```
    def forward(self, x, h):
        return x * h
               ~~~~~ <--- HERE
        self._loweredmodule.forward()
```
which shows the original source code.

iseeyuan commented on test/jit/test_backends.py:
```diff
@@ -562,6 +562,33 @@ def test_execution(self):
+        # Test exception is thrown.
+        with self.assertRaisesRegexWithHighlight(RuntimeError,
+                                                 r"node of aten::mul is not supported in this compiler", ""):
+            scripted_module_n = torch.jit.script(ErrorMessagesWithCompiler.ModuleNotSupported())
```
This line can be out of `with self.assertRaisesRegexWithHighlight(...`, so that readers know that the error is from the line of  `_jit_to_backend`.

Commit:
Update on "Python error unit tests on delegation of backend_with_compiler_demo"


Summary:
Added a test for errors that occur with a compiler, specifically when an
operator is not supported by the backend.

Test Plan:
Running python test/test_jit.py TestBackendsWithCompiler -v returns a
success.

Reviewers:
myuan

Subscribers:

Tasks:
T91991412

Tags:

Differential Revision: [D29374513](https://our.internmc.facebook.com/intern/diff/D29374513)

[ghstack-poisoned]

Commit:
Update on "Python error unit tests on delegation of backend_with_compiler_demo"


Summary:
Added a test for errors that occur with a compiler, specifically when an
operator is not supported by the backend.

Test Plan:
Running python test/test_jit.py TestBackendsWithCompiler -v returns a
success.

Reviewers:
myuan

Subscribers:

Tasks:
T91991412

Tags:

Differential Revision: [D29374513](https://our.internmc.facebook.com/intern/diff/D29374513)

[ghstack-poisoned]

Amyh11325 commented:
@Amyh11325 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29374513).

iseeyuan commented on test/jit/test_backends.py:
```diff
@@ -562,6 +562,39 @@ def test_execution(self):
+        scripted_module_n = torch.jit.script(ErrorMessagesWithCompiler.ModuleNotSupported())
+        # Test exception is thrown when lowering a module with an unsupported operator
+        with self.assertRaisesRegexWithHighlight(RuntimeError,
+#Special escape characters are replaced with '.'
```
Please format this line and the line below accordingly with indents.

iseeyuan reviewed (approved):
LGTM. Please remember to fix the inlined format issues before landing.

Commit:
Update on "Python error unit tests on delegation of backend_with_compiler_demo"


Summary:
Added a test for errors that occur with a compiler, specifically when an
operator is not supported by the backend.

Test Plan:
Running python test/test_jit.py TestBackendsWithCompiler -v returns a
success.

Reviewers:
myuan

Subscribers:

Tasks:
T91991412

Tags:

Differential Revision: [D29374513](https://our.internmc.facebook.com/intern/diff/D29374513)

[ghstack-poisoned]

Commit:
Update on "Python error unit tests on delegation of backend_with_compiler_demo"


Summary:
Added a test for errors that occur with a compiler, specifically when an
operator is not supported by the backend.

Test Plan:
Running python test/test_jit.py TestBackendsWithCompiler -v returns a
success.

Reviewers:
myuan

Subscribers:

Tasks:
T91991412

Tags:

Differential Revision: [D29374513](https://our.internmc.facebook.com/intern/diff/D29374513)

[ghstack-poisoned]

Commit:
Update on "Python error unit tests on delegation of backend_with_compiler_demo"


Summary:
Added a test for errors that occur with a compiler, specifically when an
operator is not supported by the backend.

Test Plan:
Running python test/test_jit.py TestBackendsWithCompiler -v returns a
success.

Reviewers:
myuan

Subscribers:

Tasks:
T91991412

Tags:

Differential Revision: [D29374513](https://our.internmc.facebook.com/intern/diff/D29374513)

[ghstack-poisoned]

Amyh11325 commented:
@Amyh11325 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29374513).

Commit:
Update on "Python error unit tests on delegation of backend_with_compiler_demo"


Summary:
Added a test for errors that occur with a compiler, specifically when an
operator is not supported by the backend.

Test Plan:
Running python test/test_jit.py TestBackendsWithCompiler -v returns a
success.

Reviewers:
myuan

Subscribers:

Tasks:
T91991412

Tags:

Differential Revision: [D29374513](https://our.internmc.facebook.com/intern/diff/D29374513)

[ghstack-poisoned]
